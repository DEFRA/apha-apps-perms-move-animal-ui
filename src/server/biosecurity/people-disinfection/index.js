import { QuestionPageController } from '../../common/controller/question-page-controller/question-page-controller.js'
import { PeopleDisinfectionAnswer } from '../../common/model/answer/people-disinfection/people-disinfection.js'
import { QuestionPage } from '../../common/model/page/question-page-model.js'
import { disinfectantPage } from '../disinfectant/index.js'

export class PeopleDisinfectionPage extends QuestionPage {
  urlPath = '/biosecurity/people-disinfection'
  sectionKey = 'biosecurity'
  question =
    'What measures are you taking to minimise the risk of staff working with the incoming cattle spreading contamination onto resident or other cattle?'

  questionKey = 'peopleDisinfection'
  Answer = PeopleDisinfectionAnswer

  nextPage() {
    return disinfectantPage
  }
}

export const peopleDisinfectionPage = new PeopleDisinfectionPage()

/**
 * @satisfies {ServerRegisterPluginObject<void>}
 */
export const peopleDisinfection = new QuestionPageController(
  peopleDisinfectionPage
).plugin()

/**
 * @import { ServerRegisterPluginObject } from '@hapi/hapi'
 */
