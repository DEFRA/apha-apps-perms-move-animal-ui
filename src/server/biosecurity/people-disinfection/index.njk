{% extends 'layouts/questions.njk' %}

{% block heading %}
  <div class="govuk-grid-row">
    <div class="govuk-grid-column-two-thirds">
      <h1 class="govuk-heading-l">{{ heading }}</h1>
    </div>
  </div>
{% endblock %}

{% block beforequestion %}
  <div class="govuk-grid-row">
    <div class="govuk-grid-column-two-thirds">
      <p>
        To prevent the spread of TB to the incoming cattle from any resident
        cattle, staff should have cleaning and disinfection measures in place.
      </p>

      <p>This can include:</p>
      <ul class="govuk-list govuk-list--bullet">
        <li>dedicated clothing and personal protective equipment (PPE)</li>
        <li>cleaning and disinfecting wellington boots, including foot dips</li>
        <li>
          cleaning and disinfection measures when contractors arrive and leave
        </li>
        <li>dedicated staff looking after the incoming cattle</li>
      </ul>
    </div>
  </div>
{% endblock %}
