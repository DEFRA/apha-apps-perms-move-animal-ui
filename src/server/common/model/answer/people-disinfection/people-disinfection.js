import { CheckboxAnswer } from '../checkbox/checkbox.js'

/** @import {CheckboxConfig, CheckboxData} from '../checkbox/checkbox.js' */

/**
 * @typedef {{ peopleDisinfection: CheckboxData }} PeopleDisinfectionPayload
 */

/**
 * @augments {CheckboxAnswer<PeopleDisinfectionPayload>}
 */
export class PeopleDisinfectionAnswer extends CheckboxAnswer {
  /** @type {CheckboxConfig} */
  static config = {
    payloadKey: 'peopleDisinfection',
    isPageHeading: true,
    hint: 'Only select the measures you are taking. You do not need to select them all to receive your licence.',
    options: {
      ppe: {
        label: 'Dedicated clothing and personal protective equipment (PPE)'
      },
      disinfectingBoots: {
        label: 'Cleaning and disinfecting wellington boots, including foot dips'
      },
      disinfectingOnArrivalAndDeparture: {
        label:
          'Cleaning and disinfection measures when contractor arrive and leave'
      },
      dedicatedStaff: {
        label: 'Dedicated staff looking after the incoming animals'
      },
      other: {
        label: 'Other cleaning and disinfection measures'
      }
    },
    errors: {}
  }
}
