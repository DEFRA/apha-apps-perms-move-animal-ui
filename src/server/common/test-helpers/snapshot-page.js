import { parseDocument } from '../../common/test-helpers/dom.js'
import { createServer } from '~/src/server/index.js'
import { withCsrfProtection } from './csrf.js'
/** @import { Server } from '@hapi/hapi' */

/* global expect, it, beforeAll, afterAll, describe */

/**
 * @param {string} describeMessage
 * @param {string} shouldMessage
 * @param {string} urlPath
 */
export const describePageSnapshot = (
  describeMessage,
  shouldMessage,
  urlPath
) => {
  describe(describeMessage, () => {
    /** @type {Server} */
    let server

    beforeAll(async () => {
      server = await createServer()
      await server.initialize()
    })

    afterAll(async () => {
      await server.stop({ timeout: 0 })
    })

    it(shouldMessage, async () => {
      const { payload } = await server.inject(
        withCsrfProtection({
          method: 'GET',
          url: urlPath
        })
      )

      const content =
        parseDocument(payload).querySelector('#main-content')?.innerHTML

      expect(content).toMatchSnapshot()
    })
  })
}
