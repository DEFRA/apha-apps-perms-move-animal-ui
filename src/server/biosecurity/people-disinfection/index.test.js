import { PeopleDisinfectionAnswer } from '../../common/model/answer/people-disinfection/people-disinfection.js'
import { describePageSnapshot } from '../../common/test-helpers/snapshot-page.js'
import { disinfectantPage } from '../disinfectant/index.js'
import { peopleDisinfectionPage, PeopleDisinfectionPage } from './index.js'

const sectionKey = 'biosecurity'
const question =
  'What measures are you taking to minimise the risk of staff working with the incoming cattle spreading contamination onto resident or other cattle?'
const questionKey = 'peopleDisinfection'
const view = 'common/model/page/question-page.njk'
const pageUrl = '/biosecurity/people-disinfection'

describe('PeopleDisinfectionPage', () => {
  const page = new PeopleDisinfectionPage()

  it('should have the correct urlPath', () => {
    expect(page.urlPath).toBe(pageUrl)
  })

  it('should have the correct sectionKey', () => {
    expect(page.sectionKey).toBe(sectionKey)
  })

  it('should have the correct question', () => {
    expect(page.question).toBe(question)
  })

  it('should have the correct questionKey', () => {
    expect(page.questionKey).toBe(questionKey)
  })

  it('should have the correct view', () => {
    expect(page.view).toBe(view)
  })

  it('should have the correct Answer model', () => {
    expect(page.Answer).toBe(PeopleDisinfectionAnswer)
  })

  it('nextPage should return disinfectant page', () => {
    const nextPage = page.nextPage()
    expect(nextPage).toBe(disinfectantPage)
  })

  it('should export page', () => {
    expect(peopleDisinfectionPage).toBeInstanceOf(PeopleDisinfectionPage)
  })

  describePageSnapshot({
    describes: 'peopleDisinfectionPage.content',
    it: 'should render expected response and content',
    pageUrl
  })
})
