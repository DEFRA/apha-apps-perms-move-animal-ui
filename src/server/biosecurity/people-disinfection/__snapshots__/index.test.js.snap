// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleDisinfectionPage peopleDisinfectionPage.content should render expected response and content 1`] = `
"
    <form method="POST" novalidate="">
      <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
          <input type="hidden" name="crumb" value="csrf-value">
          <input type="hidden" name="nextPage" value="">

<fieldset class="govuk-fieldset">
  <div class="govuk-form-group">
  <h1 class="govuk-label-wrapper">
    <label class="govuk-label govuk-label--l" for="peopleDisinfection">
      What measures are you taking to minimise the risk of staff working with the incoming cattle spreading contamination onto resident or other cattle?
    </label>
  </h1>
  <div id="peopleDisinfection-hint" class="govuk-hint">
    For example, using personal protective equipment (PPE) and foot dips
  </div>
  <textarea class="govuk-textarea" id="peopleDisinfection" name="peopleDisinfection" rows="8" aria-describedby="peopleDisinfection-hint"></textarea>
</div>


</fieldset>
            <button type="submit" class="govuk-button" data-module="govuk-button" id="continue-button">
  Continue
</button>
        </div>
      </div>
    </form>
        "
`;
